### PR TITLE
Prevent VAR_RNOISE from being a zeroed array.

### PR DESCRIPTION
**Pull request by mairanteodoro**
This PR updates a few unit tests to prevent `VAR_RNOISE` from being a zeroed array (such edge case might need to be addressed eventually).

For the following two unit tests, we increased the value of `im.var_rnoise` from zero to 0.01:
- `test_custom_wcs_resample_imaging`;
- `test_custom_refwcs_resample_imaging`;

Additionally, the error in `test_custom_refwcs_pixel_shape_imaging` was caused by an incorrect `pixel_shape` specification in the WCS. The test was setting `result.meta.wcs.pixel_shape = data1.shape` where `data1.shape` is in NumPy's `(ny, nx)` format (height, width), but WCS `pixel_shape` expects `(nx, ny)` format (width, height). The test now correctly verifies that:
- the WCS can use an explicit pixel_shape setting;
- the output shape matches the expected dimensions;
- the pixel area can be set correctly in the result metadata.

This PR also contains a temporary change to pyproject.toml so that it can run against the `stcal` changes listed in [PR #359](https://github.com/spacetelescope/stcal/pull/359).

## Note
This PR depends on changes in the following PR:
- stcal (https://github.com/spacetelescope/stcal/pull/359).

## Regression tests
All the regression tests are passing:
- https://github.com/spacetelescope/RegressionTests/actions/runs/17563400895

## Tasks
- [x] If you have a specific reviewer in mind, tag them.
- [x] add a build milestone, i.e. `Build 12.0` (use the [latest build](https://github.com/spacetelescope/jwst/milestones) if not sure)
- [ ] Does this PR change user-facing code / API? (if not, label with `no-changelog-entry-needed`)
  - [ ] write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see [changelog readme](https://github.com/spacetelescope/jwst/blob/main/changes/README.rst) for instructions) 
  - [ ] update or add relevant tests
  - [ ] update relevant docstrings and / or `docs/` page
  - [ ] [start a regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/jwst.yml) and include a link to the running job ([click here for instructions](https://github.com/spacetelescope/RegressionTests/blob/main/docs/running_regression_tests.md))
    - [ ] Do truth files need to be updated ("okified")?
      - [ ] **after the reviewer has approved these changes**, run `okify_regtests` to update the truth files
- [ ] if a JIRA ticket exists, [make sure it is resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

File: jwst/resample/tests/test_resample_step.py [hunk 1]
```diff
@@ -991,6 +991,7 @@ def test_resample_undefined_variance(nircam_rate, shape):
 def test_custom_wcs_resample_imaging(nircam_rate, ratio, rotation, crpix, crval, shape):
     im = AssignWcsStep.call(nircam_rate, sip_approx=False)
     im.data += 5
+    im.var_rnoise += 0.01
```
Comment (pllim) [inline jwst/resample/tests/test_resample_step.py:994]: What does this mean for the real data that is already out there? And I am guessing new pipeline of same data will have different VAR_RNOISE values; how will that affect downstream?

Comment (tapastro) [reply jwst/resample/tests/test_resample_step.py:994]: Real data read noise variance arrays are populated in ramp_fitting using the read noise arrays gathered from ref files. I don't believe the read noise should ever be zero, and the variance should always contain non-zero positive value unless the ramp slope is negative for the entire image or the reference file is incorrectly generated.

That said, this discussion indirectly points to why Roman is using var_sky instead of var_rnoise as a weighting scheme - the readnoise variance is not a sufficient descriptor for the statistical significance of a pixel flux value. Higher readnoise variance would de-weight a pixel contribution, but if the variance in the flux is dominated by poisson noise, the readnoise variance should not be causing such de-weighting. It's also why many JWST modes are now using a by-exposure weight rather than a by-pixel weight, because the results of readnoise variance weighting is quite unreliable.

Comment (mairanteodoro) [reply jwst/resample/tests/test_resample_step.py:994]: @pllim Should we mark this conversation as resolved?

Comment (pllim) [reply jwst/resample/tests/test_resample_step.py:994]: Of course. Thank you for the clarifications!

**Comment by pllim** [comment]
pre-commit.ci autofix

**Comment by pllim** [comment]
I took the liberty of asking the bot to fix pre-commit check failures. Hope you don't mind. Looking at what angered the bot, you might have to configure your editor to automatically delete trailing whitespace on save. FYI.

**Comment by mairanteodoro** [comment]
> I took the liberty of asking the bot to fix pre-commit check failures. Hope you don't mind. Looking at what angered the bot, you might have to configure your editor to automatically delete trailing whitespace on save. FYI.

Thanks, @pllim! I will set my editor to take care of that, but I thought these simple style-check fixes were being taken care of during the pre-commit phase... apparently I'm wrong.
